contracts/core/Allo.sol

```diff
@@ -29,8 +45,9 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
     /// @notice Struct to hold details of an Pool
     struct Pool {
         bytes32 identityId;
-        IAllocationStrategy allocationStrategy;
-        IDistributionStrategy distributionStrategy;
+        BaseStrategy strategy;
```

Comment:
Shouldn't we use the interface instead ? 
(This goes with having getter functions for the storage variable on BaseStrategy)

Reply:
Done.